src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -96,45 +102,233 @@ static ucs_status_t uct_cuda_ipc_iface_query(uct_iface_h iface,
-    .iface_fence              = uct_base_iface_fence,
-    .iface_progress_enable    = ucs_empty_function,
-    .iface_progress_disable   = ucs_empty_function,
-    .iface_progress           = ucs_empty_function_return_zero,
+    .iface_flush              = uct_cuda_ipc_iface_flush,
+    .iface_fence              = uct_base_iface_fence, /*TODO*/
+    .iface_progress_enable    = uct_base_iface_progress_enable,
+    .iface_progress_disable   = uct_base_iface_progress_disable,
+    .iface_progress           = uct_cuda_ipc_iface_progress,
     .iface_close              = UCS_CLASS_DELETE_FUNC_NAME(uct_cuda_ipc_iface_t),
     .iface_query              = uct_cuda_ipc_iface_query,
     .iface_get_device_address = uct_cuda_ipc_iface_get_device_address,
     .iface_get_address        = uct_cuda_ipc_iface_get_address,
     .iface_is_reachable       = uct_cuda_ipc_iface_is_reachable,
 };
 
+static void uct_cuda_ipc_event_desc_init(ucs_mpool_t *mp, void *obj, void *chunk)
+{
+    uct_cuda_ipc_event_desc_t *base = (uct_cuda_ipc_event_desc_t *) obj;
+    CUresult                  cu_ret;
+
+    memset(base, 0 , sizeof(*base));
+    cu_ret = cuEventCreate(&(base->event), CU_EVENT_DISABLE_TIMING);
+    if (CUDA_SUCCESS != cu_ret) {
+        cuGetErrorString(cu_ret, &cu_err_str);
+        ucs_error("cuEventCreate Failed ret:%s", cu_err_str);
+    }
+}
+
+static void uct_cuda_ipc_event_desc_cleanup(ucs_mpool_t *mp, void *obj)
+{
+    uct_cuda_ipc_event_desc_t *base = (uct_cuda_ipc_event_desc_t *) obj;
+    CUresult                  cu_ret;
+
+    cu_ret = cuEventDestroy(base->event);
+    if (CUDA_SUCCESS != cu_ret) {
+        cuGetErrorString(cu_ret, &cu_err_str);
+        ucs_error("cuEventDestroy Failed ret:%s", cu_err_str);
+    }
+}
+
+ucs_status_t uct_cuda_ipc_iface_init_streams(uct_cuda_ipc_iface_t *iface)
+{
+    int       i;
+    CUresult  cu_ret;
+
+    for (i = 0; i < iface->device_count; i++) {
+        cu_ret = cuStreamCreate(&iface->stream_d2d[i], CU_STREAM_NON_BLOCKING);
+        if (cu_ret != CUDA_SUCCESS) {
+            cuGetErrorString(cu_ret, &cu_err_str);
+            ucs_error("cuStreamCreate d2d (lane %d) error; ret:%s", i,
+                      cu_err_str);
+            return UCS_ERR_IO_ERROR;
+        }
+    }
+    iface->streams_initialized = 1;
+
+    return UCS_OK;
+}
+
+static ucs_mpool_ops_t uct_cuda_ipc_event_desc_mpool_ops = {
+    .chunk_alloc   = ucs_mpool_chunk_malloc,
+    .chunk_release = ucs_mpool_chunk_free,
+    .obj_init      = uct_cuda_ipc_event_desc_init,
+    .obj_cleanup   = uct_cuda_ipc_event_desc_cleanup,
+};
+
 static UCS_CLASS_INIT_FUNC(uct_cuda_ipc_iface_t, uct_md_h md, uct_worker_h worker,
                            const uct_iface_params_t *params,
                            const uct_iface_config_t *tl_config)
 {
+    uct_cuda_ipc_iface_config_t *config = NULL;
+    ucs_status_t status;
+    CUresult     cu_ret;
+    int          dev_count;
+    int          i = 0, j = 0;
+
```

Comment:
don't align variable names in the function. check for all functions.

Reply:
if you assign a value to vairble, they should declare before unassigned variables

Reply:
I've fixed all such instances in the new commits